frontend/src/metabase/core/components/EditableText/EditableText.tsx

```diff
@@ -99,21 +117,24 @@ const EditableText = forwardRef(function EditableText(
   return (
     <EditableTextRoot
       {...props}
-      ref={ref}
+      ref={rootRef}
       isEditing={isEditing}
       isDisabled={isDisabled}
       data-value={`${displayValue}\u00A0`}
     >
-      <EditableTextArea
-        value={inputValue}
-        placeholder={placeholder}
-        disabled={isDisabled}
-        data-testid={dataTestId}
-        onFocus={onFocus}
-        onBlur={handleBlur}
-        onChange={handleChange}
-        onKeyDown={handleKeyDown}
-      />
+      {children ?? (
```

Comment:
The component composition seems off. A couple of ideas:
- Extract logic from `EditableText` to a hook and use it here and in the editable markdown component, and remove this conditional logic
- Further subdivide components

Reply:
How will extracting of the logic to hook simplify our life? I see your point about lack of composition what us partially true, but our goal here is to have EditableText with support of markdown and possibility to reuse EditableText component in the future, not to reuse hooks IMO